### PR TITLE
Linux support

### PR DESCRIPTION
**Pull request by marinhekman**
Converted to a newer .net 6 framework. (Current LTS)
There are some deprecation notices however, .net 7 will stop support linux on the existing code, so it will need a lot of re-factoring, using third party libraries.
For now , we have linux support through .net , although deprected, and able to use it in docker.  

* added main try-catch to msw. will slow-down performance maybe , but at least we catch all errors outputed to console.

### CODE REVIEW

File: CEL/CEL/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+WORKDIR "/src/CEL.core"
+RUN dotnet build "CEL.core.csproj" -c Release -o /app/build
+
+FROM build AS publish
+RUN dotnet publish "CEL.core.csproj" -c Release -o /app/publish
+
+FROM base AS final
+WORKDIR /app
+COPY --from=publish /app/publish .
+ENTRYPOINT ["dotnet", "CEL.core.dll"]
```
Comment (marinhekman) [inline CEL/CEL/Dockerfile:18]: not using this I think , it was generated by either Rider/Visual Studio...

File: MEL/MEL/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+WORKDIR "/src/MEL.core"
+RUN dotnet build "MEL.core.csproj" -c Release -o /app/build
+
+FROM build AS publish
+RUN dotnet publish "MEL.core.csproj" -c Release -o /app/publish
+
+FROM base AS final
+WORKDIR /app
+COPY --from=publish /app/publish .
+ENTRYPOINT ["dotnet", "MEL.core.dll"]
```
Comment (marinhekman) [inline MEL/MEL/Dockerfile:18]: not using this

File: MSW/MSW/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+WORKDIR "/src/MSW"
+RUN dotnet build "MSW.csproj" -c Release -o /app/build
+
+FROM build AS publish
+RUN dotnet publish "MSW.csproj" -c Release -o /app/publish
+
+FROM base AS final
+WORKDIR /app
+COPY --from=publish /app/publish .
+ENTRYPOINT ["dotnet", "MSW.dll"]
```
Comment (marinhekman) [inline MSW/MSW/Dockerfile:18]: not using this

File: MSW/MSW/MSW_config.json [hunk 4]
```diff
@@ -2,19 +2,19 @@
 	"available_simulations": [
 		{
 			"simulation_name": "CEL",
-			"relative_exe_path": "../../../../CEL/CEL/bin/release/CEL.exe"
+			"relative_exe_path": "CEL"
```
Comment (marinhekman) [inline MSW/MSW/MSW_config.json:5]: we just assume (after publish) that all simulations are in the same folder...

File: REL/REL/REL_Config.json [hunk 5]
```diff
@@ -1,3 +1,3 @@
 ﻿{
-	"api_root": "http://localhost/1/"
+	"api_root": "http://localhost/dev/1/"
```
Comment (marinhekman) [inline REL/REL/REL_Config.json:2]: don't know why this is changed, I think it can be reverted...

File: SEL/SEL/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+WORKDIR "/src/SEL.core"
+RUN dotnet build "SEL.core.csproj" -c Release -o /app/build
+
+FROM build AS publish
+RUN dotnet publish "SEL.core.csproj" -c Release -o /app/publish
+
+FROM base AS final
+WORKDIR /app
+COPY --from=publish /app/publish .
+ENTRYPOINT ["dotnet", "SEL.core.dll"]
```
Comment (marinhekman) [inline SEL/SEL/Dockerfile:18]: not used at the moment

File: build.bat [hunk 7]
```diff
@@ -0,0 +1,153 @@
+@echo off
```
Comment (marinhekman) [inline build.bat:1]: aah, I like this build script, to just build all for multiple targets and output it to the msp server simulations folder. Of course given the correct arguments. :)

File: CEL/CEL/CEL_config.json [hunk 8]
```diff
@@ -1,3 +1,3 @@
 {
-   "api_root": "http://localhost/5/"
+   "api_root": "http://localhost/1/"
```
Comment (hwarmelink) [inline CEL/CEL/CEL_config.json:2]: This is a little strange right? A hardcoded reference to the very first game session only, created in the server manager? Or is this variable not used or altered later?

**Review by hwarmelink — APPROVED** [review]

File: CEL/CEL/CEL_config.json (hunk 8 above)
Comment (hwarmelink) [reply CEL/CEL/CEL_config.json:2]: it's not used :)

**Pull request merged**